pkg/aks/create.go

```diff
@@ -380,7 +380,7 @@ func hasHTTPApplicationRoutingSupport(spec *aksv1.AKSClusterConfigSpec) bool {
 
 func hasAgentPoolProfile(name *string, agentPoolProfiles *[]containerservice.ManagedClusterAgentPoolProfile) bool {
 	for _, ap := range *agentPoolProfiles {
-		if ap.Name == name {
```

Comment:
I don't think so. If an agent pool profile exists it will have a name that was either created through rancher or auto generated in AKS (usually named "agentpool")